### PR TITLE
Declare explicit type for configuration

### PR DESCRIPTION
**Pull request by jessitron**
Depends on https://github.com/atomist/sdm/pull/679

### CODE REVIEW

File: lib/k8s.ts [hunk 1]
```diff
@@ -35,6 +36,18 @@ export interface SdmPackK8sOptions {
     addCommands?: boolean;
 }
 
+export interface K8sConfiguration {
+    k8s?: {
+        app?: KubernetesApplication;
```
Comment (ddgenome) [inline lib/k8s.ts]: This should probably be `Partial<KubernetesApplication>`.

File: lib/deploy/data.ts [hunk 2]
```diff
@@ -152,7 +153,9 @@ export async function defaultKubernetesApplication(
     context: HandlerContext,
 ): Promise<KubernetesApplication> {
 
-    const configAppData: Partial<KubernetesApplication> = _.get(k8Deploy, "sdm.configuration.sdm.k8s.app", {});
+    const k8sConfig = (k8Deploy.sdm.configuration.sdm.k8s || {}) as K8sConfiguration["k8s"];
+
+    const configAppData: Partial<KubernetesApplication> = k8sConfig.app || {};
```
Comment (ddgenome) [inline lib/deploy/data.ts]: If you make app a `Partial<KubernetesApplication>`, then you do not need to have the type here.

File: lib/deploy/data.ts [hunk 3]
```diff
@@ -152,7 +153,9 @@ export async function defaultKubernetesApplication(
     context: HandlerContext,
 ): Promise<KubernetesApplication> {
 
-    const configAppData: Partial<KubernetesApplication> = _.get(k8Deploy, "sdm.configuration.sdm.k8s.app", {});
+    const k8sConfig = (k8Deploy.sdm.configuration.sdm.k8s || {}) as K8sConfiguration["k8s"];
```
Comment (ddgenome) [inline lib/deploy/data.ts]: This is less safe, unless `k8Deploy.sdm.configuration.sdm` is always defined.

**Comment by jessitron** [comment]
I lost this, and now I don't remember it, and it isn't worth redoing.

**Pull request closed without merging**